### PR TITLE
range_n_clusters added to elbow_curve signature

### LINKED ISSUE #148: inconsistent naming `silhouette_analysis` and `elbow_curve`
`silhouette_analysis` and `elbow_curve` take a range of clusters, but they're named differently.

`range_n_clusters` in  `silhouette_analysis` and `n_clusters` in `elbow_curve`.  `range_n_clusters`  is a more accurate description we should rename the argument in `elbow_curve`.

since this is an API-breaking change we should keep both for a few releases and do some validations. if `n_clusters` does not have the default value, show a FutureWarning saying that they should switch to the new `range_n_clusters`. Throw an error if they're passing both `n_clusters` and `range_n_clusters`, telling them to only use `range_n_clusters`

### PR DESCRIPTION
**Pull request by yafimvo**
## Describe your changes
1. `range_n_clusters` added to elbow_curve signature
2. elbow_curve still supports `n_clusters`
3. Deprecation warning added
4. AttributeError added when using `range_n_clusters` and `n_clusters` at the same time
5. Examples updated

## Issue ticket number and link
Closes #148 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests (when necessary).
- [x] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

File: src/sklearn_evaluation/plot/clustering.py [hunk 1]
```diff
@@ -78,28 +87,47 @@ def elbow_curve(X, clf, n_clusters=None, n_jobs=1, show_cluster_time=True, ax=No
     ax: matplotlib Axes
         Axes containing the plot
 
+    Notes
+    -----
+    .. deprecated:: 0.8.7
```
Comment (idomic) [inline src/sklearn_evaluation/plot/clustering.py:92]: We probably want to give users a heads up of minor or 2,
Did you check the guide about deprecation warnings?

Comment (idomic) [reply src/sklearn_evaluation/plot/clustering.py:92]: change to 0.8.9, besides that looks good.

Comment (idomic) [reply src/sklearn_evaluation/plot/clustering.py:92]: [guide](https://github.com/ploomber/contributing/blob/main/documentation/README.md#new-features-and-breaking-api-changes)

Comment (yafimvo) [reply src/sklearn_evaluation/plot/clustering.py:92]: done

Comment (edublancas) [reply src/sklearn_evaluation/plot/clustering.py:92]: the lin Ido shared is outdated, this is the guide for deprecations: https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-renamed 

> change to 0.8.9, besides that looks good.

the guide also includes an example of the docstring, we use `.. versionchanged::` for parameter renaming

> We probably want to give users a heads up of minor or 2,

it's actually a period of two major releases. so if we deprecate this in the next release (0.8.7), we'll remove it in 0.10

File: src/sklearn_evaluation/plot/clustering.py [hunk 2]
```diff
@@ -78,28 +87,47 @@ def elbow_curve(X, clf, n_clusters=None, n_jobs=1, show_cluster_time=True, ax=No
...
+        'n_clusters' renamed to 'range_n_clusters' and will be removed in version 0.8.9
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/clustering.py:93]: this shouldn't be 0.8.9, it should be 0.9. we're following semantic versioning and breaking API changes only happen in major releases

File: src/sklearn_evaluation/plot/clustering.py [hunk 3]
```diff
@@ -78,28 +87,47 @@ def elbow_curve(X, clf, n_clusters=None, n_jobs=1, show_cluster_time=True, ax=No
...
     --------
     .. plot:: ../examples/elbow_curve.py
 
     """
-    if n_clusters is None:
-        n_clusters = range(1, 10, 2)
+
+    if n_clusters is not None:
+        if range_n_clusters is not None:
+            raise AttributeError(
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/clustering.py:103]: deprecations should show warnings, not throw exceptions, follow  the guidelines from here https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-renamed

Comment (idomic) [reply src/sklearn_evaluation/plot/clustering.py:103]: @yafimvo fix those 2 in a new PR

Comment (yafimvo) [reply src/sklearn_evaluation/plot/clustering.py:103]: @edublancas in the [original issue](https://github.com/ploomber/sklearn-evaluation/issues/148) you wrote to throw an error when users pass both `n_clusters` and `range_n_clusters`, telling them to only use `range_n_clusters`.

should I remove this error and throw a warning instead?

Comment (edublancas) [reply src/sklearn_evaluation/plot/clustering.py:103]: sorry for the back and forth. I had forgotten I said we'd raise an exception. I discussed this with Ido yesterday. the idea is to use this [guide](https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-renamed) for all deprecations - it contains functions to encapsulate the process. I'm gonna make some changes so let's put this on hold for a bit

Comment (idomic) [reply src/sklearn_evaluation/plot/clustering.py:103]: There's also a parameter that should be renamed from `labels` to `label`

**Pull request merged**